### PR TITLE
Feature/markhornak/sc 106148/implement get folders

### PR DESCRIPTION
**Pull request by markhornak**
[Shortuct](https://app.shortcut.com/guru/story/106148/implement-get-folders)

### Overview
 
Update to py_sdk to add get_folders() call.  This included adding a new class Folder to contain the objects returned by the get_folders call.  Test coverage was also added.

**Note**: it seems this PR contains some massive formatting changes that came about from VSCodes use of an updated python formatter, apologies if this is hard to discern...

### CODE REVIEW

File: guru/__init__.py [hunk 1]
```diff
@@ -1,53 +1,54 @@
...
 )
 
 # you might need these to check if an item on a board is
 # an instance of Section or Card.
 from guru.data_objects import (
-  Board,
-  BoardGroup,
-  Card,
-  Section
+    Board,
```
Comment (jahlbornG) [inline guru/__init__.py:33]: it looks like you changed the indentation level of the project from 2 to 4.  was this necessary?  caused a lot of unnecessary churn in the project and probably makes these files inconsistent with other files in the project now.

Comment (markhornak) [reply guru/__init__.py:33]: VSCode's python formatter did that...but only noticed once it was pushed up with the diff.  I'll see if I can fix that real quick.

Comment (markhornak) [reply guru/__init__.py:33]: fixed here : [cafc668](https://github.com/guruhq/py-sdk/pull/52/commits/cafc6684140baf66c7c567d5da73496306788f93)

File: guru/data_objects.py [hunk 2]
```diff
@@ -48,16 +50,52 @@ def __init__(self, data, guru=None):
...
+  - `title` is the folder's name as it's displayed in the UI.
+  - `description` is an optional description.
+  - `id` is the folder's internal ID.
+  - `url` is the full URL for the folder.
+  - `items` is the list of items on the folder, where each item can be a Card object or a Folder object.
+  - `cards` is a flattened list of Card objects for each card on the folder.
+  - `folders` is the list of Section objects for each section on the board.
+  - `all_items` is a flattened list where each is a Card object or Section object. This is similar to the `items` list
+    except a board with 2 sections (each containing 1 card) will have two items (just the sections) but `all_items` will
+    have all four items.
```
Comment (jahlbornG) [inline guru/data_objects.py:79]: is this description accurate?

File: guru/core.py [hunk 3]
```diff
@@ -2151,33 +2199,62 @@ def get_board(self, board, collection=None, board_group=None, cache=True):
...
     return Board(response.json(), guru=self)
-  
+
+  def get_folders(self, collection=None, folder=None, cache=False):
+    """
+                    Gets a list of folders you can see. You can optionally filter by collection.
+
+    Args:
+            collection (str or Collection, optional): The name or ID of a collection or a Collection object
+                    to filter by. If this is not provided, you'll get back a list of all boards in all collections
```
Comment (jahlbornG) [inline guru/core.py:2218]: nit, "boards -> "folders"

File: guru/data_objects.py (hunk 2 above)
Comment (markhornak) [reply guru/data_objects.py]: fixed here: [ef2f351](https://github.com/guruhq/py-sdk/pull/52/commits/ef2f3514a7e97ca8efec60b93c5fd3ada4e62b31)

File: guru/core.py (hunk 3 above)
Comment (markhornak) [reply guru/core.py]: fixed here : [ef2f351](https://github.com/guruhq/py-sdk/pull/52/commits/ef2f3514a7e97ca8efec60b93c5fd3ada4e62b31)

File: guru/core.py [hunk 4]
```diff
@@ -2151,33 +2199,62 @@ def get_board(self, board, collection=None, board_group=None, cache=True):
...
+
+    Returns:
+            list of Folder: Either all folders you have access to or all folders within the specified collection.
+
+    """
+    # optional filter by collection.
+    if collection:
+      collection_obj = self.get_collection(collection, cache=True)
+      if not collection_obj:
+        self.__log(make_red("colud not find collection:", collection))
```
[the remark this answers is not in the document]
Comment (markhornak) [reply guru/core.py]: fixed here: [ef2f351](https://github.com/guruhq/py-sdk/pull/52/commits/ef2f3514a7e97ca8efec60b93c5fd3ada4e62b31)

File: guru/data_objects.py [hunk 5]
```diff
@@ -48,16 +50,52 @@ def __init__(self, data, guru=None):
...
+  """
+
+  def __init__(self, data, guru=None, home_board=None):
```
[the remark this answers is not in the document]
Comment (markhornak) [reply guru/data_objects.py]: fixed here: [ef2f351](https://github.com/guruhq/py-sdk/pull/52/commits/ef2f3514a7e97ca8efec60b93c5fd3ada4e62b31)

**Comment by markhornak** [comment]
> 

@aHojo - I can't seem to find any of your comments to review or resolve them!!

**Review by jahlbornG — APPROVED** [review]

**Pull request closed without merging**